RoboJackets/igvc-software
# GPS Coordinates Fix

# Description

Fixes bug with GPS coordinates produced by the hector gazebo plugin. I think this bug was the root cause of #492 and #529 managed to patch it by adding an offset to the IMUs.

This PR does the following:
- Fixes bug with GPS coordinates
- Fixes all waypoint files whose coordinates were previously wrong due to the bug.
- Changes Swervi and Jessi IMU and Magnetometer yaw orientation to be 0 when facing north since most IMUs report 0 when facing north. I looked at the datasheet of the IMUs we currently have (Hillcrest Labs FSM-9) but couldn't find if this is actually true. Will look at this again during testing.
- Fixes `robot_localization` node by setting the correct yaw offset.

Fixes #874 

# Testing steps
## Test Case 1
1. Run `roslaunch igvc_gazebo qualification.launch track:=0` (tracks 0 through 10 available to test)
2. Run `roslaunch igvc_navigation navigation_simulation.launch`

Expectation: Robot navigates to correct locations in simulation as expected.

## Test Case 2
1. Run `roslaunch igvc_gazebo qualification.launch track:=0` (tracks 0 through 10 available to test)
2. Run `rostopic echo /fix`
3. Move robot in +X axis and observe longitude increase.
4. Move robot in +Y axis and observe latitude increase.

Expectation: Latitude and longitude increase as expected in respective axes.

# Self Checklist
